### PR TITLE
Add host fallback when nvjpegDecodeJpegDevice fails

### PR DESCRIPTION
**Pull request by JanuszL**
- in some cases, nvjpegDecodeJpegDevice can fail (like unusual chroma sampling), so adding a host fallback, in that case, would allow decoding of the input image

    Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds host fallback when nvjpegDecodeJpegDevice fails

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     in some cases, nvjpegDecodeJpegDevice can fail (like unusual chroma sampling), so adding a host fallback, in that case, would allow decoding of the input image
 - Affected modules and functionalities:
     nvjpeg_decoder_decoupled_api.h
 - Key points relevant for the review:
     NA
 - Validation and testing:
     New test data has been added - https://github.com/NVIDIA/DALI_extra/pull/52
 - Documentation (including examples):
     NA

Relates to https://github.com/NVIDIA/DALI/issues/2798
**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

File: dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h [hunk 1]
```diff
@@ -836,8 +836,16 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
...
                      file_name);
 
-      NVJPEG_CALL_EX(nvjpegDecodeJpegDevice(handle_, decoder, state, &nvjpeg_image, stream),
-                     file_name);
+      ret = nvjpegDecodeJpegDevice(handle_, decoder, state, &nvjpeg_image, stream);
+      // if nvJPEG fails try HostDecoder
+      if (ret != NVJPEG_STATUS_SUCCESS) {
+        auto warning_msg = make_string("NVJPEG error \"", static_cast<int>(ret), "\" : ",
+                           nvjpeg_parse_error_code(ret), " ", file_name);
+        DALI_WARN(warning_msg);
```
Comment (JanuszL) [inline dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h:844]: I'm not sure about this, but maybe we should warn as fallback to CPU at this stage is not expected and indicates a problem with nvJPEG we should see and not silently ignore.

**Comment by JanuszL** [comment]
!build

**Review by awolant — APPROVED** [review]

**Review by jantonguirao — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Review by awolant — APPROVED** [review]

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**